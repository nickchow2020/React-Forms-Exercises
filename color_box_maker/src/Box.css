.box{
    margin: .5rem;
}