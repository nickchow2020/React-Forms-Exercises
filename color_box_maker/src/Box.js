import React from "react"
import "./Box.css"
const Box = ({width,height,backgroundColor,id,deleteBox})=>{


    const handleClick = ()=>{
        deleteBox(id)
    }

    return(
        <div>
            <div 
            className="box" 
            id={id}
            style={{
                width:`${width}px`,
                height:`${height}px`,
                backgroundColor}}>
            </div>
            <button onClick={handleClick} data-testid="remove">X</button>
        </div>
    )
}


export default Box;