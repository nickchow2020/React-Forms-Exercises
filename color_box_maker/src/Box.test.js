import { render} from '@testing-library/react';
import Box from './Box';


test("should render component",()=>{
    render(<Box />)
})


test("should match snapshot",()=>{
    const {asFragment} = render(<Box />)    
    expect(asFragment()).toMatchSnapshot()
})

test("should match snapshot with pros",()=>{
    const {asFragment} = render(<Box width="10" height="10" backgroundColor="tomato" id="100"/>)    
    expect(asFragment()).toMatchSnapshot()
})


test("should have remove button",()=>{
    const {getByTestId,getByText} = render(<Box width="10" height="10" backgroundColor="tomato" id="100"/>)    
    
    const removeBtn = getByTestId("remove")

    expect(removeBtn).toBeInTheDocument()
    expect(getByText("X")).toBeInTheDocument()
})


