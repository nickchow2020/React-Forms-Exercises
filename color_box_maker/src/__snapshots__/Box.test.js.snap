// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="box"
    />
    <button
      data-testid="remove"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`should match snapshot with pros 1`] = `
<DocumentFragment>
  <div>
    <div
      class="box"
      id="100"
      style="width: 10px; height: 10px; background-color: tomato;"
    />
    <button
      data-testid="remove"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
